tests/models/test_base.py

```diff
@@ -67,42 +68,42 @@ def test_fileondisk_save_failuremodes( diskfile ):
-        diskfile.save( data1 )
-    with pytest.raises( FileNotFoundError ):
-        ifp = open( diskfile.get_fullpath(), 'rb' )
-        ifp.close()
-    diskfile.filepath_extensions = [ '1', '2', '3' ]
-    with pytest.raises( RuntimeError, match=r'Data integrity error; len\(md5sum_extensions\).*' ):
-        diskfile.save( data1, extension='1' )
-    with pytest.raises( FileNotFoundError ):
-        ifp = open( diskfile.get_fullpath()[0], 'rb' )
-        ifp.close()
-    with pytest.raises( FileNotFoundError ):
-        ifp = open( diskfile.get_fullpath()[1], 'rb' )
-        ifp.close()
-    diskfile.md5sum_extensions = None
-    diskfile.filepath_extensions = [ '1', '2' ]
-    with pytest.raises( RuntimeError, match='Data integrity error; filepath_extensions is not null.*' ):
-        diskfile.save( data1, extension='1' )
-    with pytest.raises( FileNotFoundError ):
-        ifp = open( diskfile.get_fullpath()[0], 'rb' )
-        ifp.close()
-    with pytest.raises( FileNotFoundError ):
-        ifp = open( diskfile.get_fullpath()[1], 'rb' )
-        ifp.close()
-    diskfile.filepath_extensions = None
+    # diskfile.filepath_extensions = [ '1', '2' ]
+    # with pytest.raises( RuntimeError, match='Tried to save a non-extension file, but this file has extensions' ):
+    #     diskfile.save( data1 )
+    # with pytest.raises( FileNotFoundError ):
+    #     ifp = open( diskfile.get_fullpath()[0], 'rb' )
+    #     ifp.close()
+    # with pytest.raises( FileNotFoundError ):
+    #     ifp = open( diskfile.get_fullpath()[1], 'rb' )
+    #     ifp.close()
+    # diskfile.filepath_extensions = None
+    # diskfile.md5sum_extensions = [ uuid.uuid4(), uuid.uuid4() ]
+    # with pytest.raises( RuntimeError, match='Data integrity error; filepath_extensions is null.*' ):
+    #     diskfile.save( data1 )
+    # with pytest.raises( FileNotFoundError ):
+    #     ifp = open( diskfile.get_fullpath(), 'rb' )
+    #     ifp.close()
+    # diskfile.filepath_extensions = [ '1', '2', '3' ]
+    # with pytest.raises( RuntimeError, match=r'Data integrity error; len\(md5sum_extensions\).*' ):
+    #     diskfile.save( data1, extension='1' )
+    # with pytest.raises( FileNotFoundError ):
+    #     ifp = open( diskfile.get_fullpath()[0], 'rb' )
+    #     ifp.close()
+    # with pytest.raises( FileNotFoundError ):
+    #     ifp = open( diskfile.get_fullpath()[1], 'rb' )
+    #     ifp.close()
+    # diskfile.md5sum_extensions = None
+    # diskfile.filepath_extensions = [ '1', '2' ]
+    # with pytest.raises( RuntimeError, match='Data integrity error; filepath_extensions is not null.*' ):
+    #     diskfile.save( data1, extension='1' )
+    # with pytest.raises( FileNotFoundError ):
+    #     ifp = open( diskfile.get_fullpath()[0], 'rb' )
+    #     ifp.close()
+    # with pytest.raises( FileNotFoundError ):
+    #     ifp = open( diskfile.get_fullpath()[1], 'rb' )
+    #     ifp.close()
+    # diskfile.filepath_extensions = None
```

Comment:
These should probably be replaced with tests that make sure that the keys of the md5sum extensions match what's listed in the filepath extensions.  The checks that the _extensions fileds are None when there are no file extensions should stay in.

Reply:
I think the reason I removed these tests (for now) is that there are situations where you'd call save() without having a full match between the extensions and the md5sums because you need to call save() for each extension. 
If you don't allow them to be different at any point, you may not be able to do that in sequence. I need to have another look at save() logic to see which test is actually needed.